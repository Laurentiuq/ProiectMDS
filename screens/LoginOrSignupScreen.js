import { KeyboardAvoidingView, StyleSheet, Text, View, TextInput, TouchableOpacity, Image, ImageBackground } from 'react-native'
import React, { useEffect, useState } from 'react'
import { useNavigation } from '@react-navigation/native';
import { getAuth } from 'firebase/auth'
export default function LoginOrSignupScreen() {

    // get the navigation object from useNavigation hook
    const navigation = useNavigation();


    // Redirect to HomeScreen if user logged in
    useEffect(() => {
        const auth = getAuth();
        const unsubcribe =
            auth.onAuthStateChanged((user) => {
                if (user) {
                    navigation.replace('Home');
                }
            })
        return unsubcribe;
    }, [])

    // GO to register screen when press register button
    const goToRegister = () => {
        navigation.navigate('Register');
    }

    // Go to Login screen when press register button
    const goToLogin = () => {
        navigation.navigate('Login');

    }
    return (
        <View
            style={styles.container}
            behavior="padding">

            <ImageBackground source={require("../assets/mainBackground.png")} style={styles.imageBackground}>

                <View style={styles.images}>
                    <Image source={require('../assets/AppLogo.png')} style={styles.logo} />
                    <View style={{ height: 16 }} />
                    <Image source={require('../assets/persons.png')} style={styles.persons} resizeMode="center" />
                    <View style={{ height: 32 }} />
                    <View style={{ height: 32 }} />
                </View>
                <View style={{ height: 64 }} />
                <View style={styles.squareShape}>
                    <View style={styles.buttonContainer}>
                        <Text style={{ fontWeight: 'bold', fontSize: 20, textAlign: 'center' }}>
                            Login or Sign Up</Text>
                        <View style={{ height: 8 }} />
                        <Text style={{ fontWeight: '300', fontSize: 14, textAlign: 'center' }}>
                            Login or create an account to take quiz, take part in challenge, and more.</Text>
                    </View>
                    <View style={{ height: 16 }} />
                    <View style={styles.buttonContainer}>
                        <TouchableOpacity
                            onPress={goToLogin}
                            style={styles.button}>
                            <Text style={styles.buttonText}>Login</Text>
                        </TouchableOpacity>
                    </View>
                    <View style={styles.buttonContainer}>
                        <TouchableOpacity
                            onPress={goToRegister}
                            style={[styles.button, styles.buttonOutline]}>
                            <Text style={styles.buttonOutlineText}>Create an account</Text>
                        </TouchableOpacity>
                    </View>

                </View>

            </ImageBackground>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    imageBackground: {
        flex: 1,
        width: '100%',
        justifyContent: 'center',
        alignItems: 'center',
    },

    images: {
        justifyContent: 'center',
        alignItems: 'center',
    },

    squareShape: {
        borderRadius: 20,
        padding: 20,
        width: '80%',
        display: 'flex',
        alignItems: 'flex-start',
        backgroundColor: 'white'

    },

    buttonContainer: {
        width: '100%',
        alignContent: 'center',
        marginTop: 20,

    },

    button: {
        backgroundColor: '#F16956',
        width: '100%',
        padding: 15,
        borderRadius: 20,
        alignItems: 'center',
    },
    buttonOutline: {
        backgroundColor: '#E6E6E6',
        marginTop: 5,
        // borderColor: 'rgb(241, 105, 86)',
        // borderWidth: 2,
    },
    buttonText: {
        color: 'white',
        fontWeight: '800',
        fontSize: 16,
    },
    buttonOutlineText: {
        color: 'rgb(241, 105, 86)',
        fontWeight: '800',
        fontSize: 16,
    },
    logo: {
        width: 92,
        height: 92,
  
    },
    persons: {
        height: 92,
 
    },


})