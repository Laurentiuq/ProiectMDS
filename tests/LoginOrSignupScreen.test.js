import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import LoginOrSignupScreen from '../screens/LoginOrSignupScreen.js';





jest.mock('@react-navigation/native', () => {
    return {
        useNavigation: jest.fn(() => ({ navigate: jest.fn(), replace: jest.fn() })),
    };
});

jest.mock('firebase/auth', () => {
  return {
    getAuth: jest.fn(() => ({
      onAuthStateChanged: jest.fn(),
    })),
  };
});

describe('LoginOrSignupScreen', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<LoginOrSignupScreen />);
        expect(toJSON()).toMatchSnapshot();
    });


    it('navigates to Login screen on button press', () => {
        const mockNavigation = { navigate: jest.fn(), replace: jest.fn() };
        jest.spyOn(require('@react-navigation/native'), 'useNavigation').mockReturnValue(mockNavigation);
        const { getByTestId } = render(<LoginOrSignupScreen />);
        fireEvent.press(getByTestId('login-button'));
        expect(mockNavigation.navigate).toHaveBeenCalledWith('Login');
    });



    it('navigates to Register screen on button press', () => {
        const mockNavigation = { navigate: jest.fn(), replace: jest.fn() };
        jest.spyOn(require('@react-navigation/native'), 'useNavigation').mockReturnValue(mockNavigation);
        const { getByTestId } = render(<LoginOrSignupScreen />);
        fireEvent.press(getByTestId('register-button'));
        expect(mockNavigation.navigate).toHaveBeenCalledWith('Register');
    });
});
