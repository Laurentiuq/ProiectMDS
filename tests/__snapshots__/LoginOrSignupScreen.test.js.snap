// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginOrSignupScreen renders correctly 1`] = `
<View
  behavior="padding"
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/mainBackground.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/AppLogo.png",
          }
        }
        style={
          {
            "height": 92,
            "width": 92,
          }
        }
      />
      <View
        style={
          {
            "height": 16,
          }
        }
      />
      <Image
        resizeMode="center"
        source={
          {
            "testUri": "../../../assets/persons.png",
          }
        }
        style={
          {
            "height": 92,
          }
        }
      />
      <View
        style={
          {
            "height": 32,
          }
        }
      />
      <View
        style={
          {
            "height": 32,
          }
        }
      />
    </View>
    <View
      style={
        {
          "height": 64,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "flex-start",
          "backgroundColor": "white",
          "borderRadius": 20,
          "display": "flex",
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <View
        style={
          {
            "alignContent": "center",
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Login or Sign Up
        </Text>
        <View
          style={
            {
              "height": 8,
            }
          }
        />
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "300",
              "textAlign": "center",
            }
          }
        >
          Login or create an account to take quiz, take part in challenge, and more.
        </Text>
      </View>
      <View
        style={
          {
            "height": 16,
          }
        }
      />
      <View
        style={
          {
            "alignContent": "center",
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F16956",
              "borderRadius": 20,
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
          testID="login-button"
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "800",
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignContent": "center",
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#E6E6E6",
              "borderRadius": 20,
              "marginTop": 5,
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
          testID="register-button"
        >
          <Text
            style={
              {
                "color": "rgb(241, 105, 86)",
                "fontSize": 16,
                "fontWeight": "800",
              }
            }
          >
            Create an account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
